buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -61,14 +62,12 @@
 
     private CompositeByteBuf(ByteBufAllocator alloc, boolean direct, int maxNumComponents, int initSize) {
         super(AbstractByteBufAllocator.DEFAULT_MAX_CAPACITY);
-        if (alloc == null) {
-            throw new NullPointerException("alloc");
-        }
         if (maxNumComponents < 1) {
             throw new IllegalArgumentException(
                     "maxNumComponents: " + maxNumComponents + " (expected: >= 1)");
         }
-        this.alloc = alloc;
+
+        this.alloc = ObjectUtil.checkNotNull(alloc, "alloc");
```

Comment:
@carryxyh lets move this check (and assignment) above the `maxNumComponents` check again (just like before).